File: Week 01/id_722/26-Remove-Duplicates-from-Sorted-Array.go
package main
import "fmt"
func main() {
    nums := []int {1,1,2}
	l := removeDuplicates(nums)
	fmt.Println("#v", nums)
	fmt.Println(l)
}

func removeDuplicates(nums []int) int {
	var l int
	for i := 0; i < len(nums); i++ {
		if i>0 {
			if nums[i] == nums[i-1] {
				//删除nums[nums[i+1]]
				nums = append(nums[:i],nums[i+1:]...)
				i--
			}
		}
		l = i
	}
	return l+1
}

File: Week 01/id_242/26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        i = 0
        for j in range(len(nums)):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = nums[j]
        return i + 1

    # def removeDuplicates(self, nums: List[int]) -> int:
    #     if len(nums) == 0:
    #         return 0
    #     targetIndex, OG = 1, nums[0]
    #     for i in range(1, len(nums)):
    #         if nums[i] != OG:
    #             OG = nums[i]
    #             nums[targetIndex] = nums[i]
    #             targetIndex += 1
    #     return targetIndex

        
# @lc code=end

